### PR TITLE
Tcg deck data access

### PR DESCRIPTION
**Pull request by Blackshadow8910**
Added some classes and functionality for implementing later features, including:

- SellListing
- Conversation
- Message
- deck saving and reading for test object

### CODE REVIEW

File: src/main/java/entity/Conversation.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    private final ArrayList<String> participants = new ArrayList<>();
+    private final ArrayList<Message> messages = new ArrayList<>();
+
+
+
+    public Conversation() {
+
+    }
+
+    public List<String> getParticipants() {
```
Comment (yujusti) [inline src/main/java/entity/Conversation.java:20]: Why use string?

File: src/main/java/entity/Conversation.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package entity;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class Conversation {
```
Comment (yujusti) [inline src/main/java/entity/Conversation.java:7]: is the messaging class going to be in a distinct panel from the rest?

File: src/main/java/entity/User.java [hunk 3]
```diff
@@ -1,12 +1,42 @@
...
+
+    public List<Card> getOwnedCards() {
+        return ownedCards;
+    }
+
+    public void addOwnedCard(Card card) {
+        ownedCards.add(card);
+    }
+
+    public void removeOwnedCard(String id) {
```
Comment (yujusti) [inline src/main/java/entity/User.java:33]: how can this be implemented within the user interface

File: src/main/java/data_access/database/PostGreSQLAccessObject.java [hunk 4]
```diff
@@ -97,7 +102,38 @@ public void registerUser(String username, String password) throws UserAlreadyExi
...
         try {
-            executeQuery("INSERT INTO users (username, password) VALUES ('%s', '%s');");
+            executeQuery("INSERT INTO users (username, password) VALUES ('%s', '%s');".formatted(username, password));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public Deck getDeck(String id) {
```
Comment (yujusti) [inline src/main/java/data_access/database/PostGreSQLAccessObject.java:112]: nice implementation of get deck query

**Review by yujusti — APPROVED** [review]
Overall very good implementation, I will work on the user interfaces for these things!

File: src/main/java/entity/Conversation.java (hunk 2 above)
Comment (Blackshadow8910) [reply src/main/java/entity/Conversation.java:7]: Probably, yes, if we even get around to implementing it

**Pull request merged**